lib/extensions/ar_virtual.rb

```diff
@@ -143,6 +143,7 @@ def define_delegate(method_name, method, to: nil, allow_nil: nil, default: nil)
       if allow_nil
         method_def = [
           "def #{method_name}(#{definition})",
+          "return self[:#{method_name}] if has_attribute?(:#{method_name})",
```

Comment:
Looks like it was originally a HEREDOC:

https://github.com/rails/rails/commit/2ee6229

Looks like Aaron changed it here:

https://github.com/rails/rails/commit/44c51fc9

To fix https://github.com/rails/rails/issues/13724